src/model-helpers/User.js

```diff
@@ -0,0 +1,180 @@
+import {
+  buildThing,
+  createThing,
+  setThing,
+  saveSolidDatasetAt,
+  getSolidDataset,
+  createSolidDataset,
+  getThingAll,
+  getDatetime,
+  getPodUrlAll,
+  getStringNoLocale,
+  getUrl,
+  getThing
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '../constants';
+
+import { setDocAclForUser } from '../utils/network/session-helper';
+
+/**
+ * @typedef {import('@inrupt/solid-ui-react').SessionContext} Session
+ */
+
+/**
+ * Function that updates a user's last active time on Solid Pod
+ *
+ * @memberof User
+ * @function updateUserActivity
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {URL} podUrl - the url of the user's pod to update
+ * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
+ */
+export const updateUserActivity = async (session, podUrl) => {
+  const activityDocUrl = `${podUrl}public/active.ttl`;
+  const accessObject = {
+    read: true,
+    append: true,
+    write: true,
+    control: true
+  };
+  const publicAccessObject = {
+    read: true,
+    append: false,
+    write: false,
+    control: false
+  };
+  let activityDataset;
+
+  const saveActivity = async (ttlFile) => {
+    activityDataset = setThing(activityDataset, ttlFile);
+
+    await saveSolidDatasetAt(activityDocUrl, activityDataset, {
+      fetch: session.fetch
+    });
+  };
+
+  try {
+    activityDataset = await getSolidDataset(activityDocUrl, { fetch: session.fetch });
+    const activeTTLThing = getThing(activityDataset, `${activityDocUrl}#active`);
+
+    const activeTTL = buildThing(activeTTLThing)
+      .setDatetime(RDF_PREDICATES.dateModified, new Date())
+      .build();
+
+    await saveActivity(activeTTL);
+  } catch {
+    const newActiveTTL = buildThing(createThing({ name: 'active' }))
+      .addDatetime(RDF_PREDICATES.dateModified, new Date())
+      .build();
+
+    activityDataset = createSolidDataset();
+    await saveActivity(newActiveTTL);
+
+    await setDocAclForUser(
+      session,
+      activityDocUrl,
+      'create',
+      session.info.webId,
+      accessObject,
+      publicAccessObject
+    );
+  }
+};
+
+/**
+ * Fetches the last time a user was active in pass
+ *
+ * @memberof User
+ * @function loadUserActivity
+ * @param {URL} podUrl - Url of user's pod to fetch from
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {Promise<string>} last time the user was active in PASS
+ */
+const loadUserActivity = async (podUrl, session) => {
+  const activityUrl = `${podUrl}public/active.ttl`;
+  try {
+    const solidDataset = await getSolidDataset(activityUrl, {
+      fetch: session.fetch
+    });
+    const activeThing = getThingAll(solidDataset)[0];
```

Comment:
Think we can use `getThing(activityDataset, `${activityDocUrl}#active`);` like in Line 58 in place of solidDataset since we know the exact path to the active.ttl Thing.